packages/app-mobile/components/NoteEditor/MarkdownToolbar.tsx

```diff
@@ -0,0 +1,577 @@
+		content = (
+			<Text style={{ ...styleSheet.text, ...activatedTextStyle, ...disabledTextStyle }}>
+				{spec.icon}
+			</Text>
+		);
+	} else {
+		content = spec.icon;
+	}
+
+	const onPress = useCallback(() => {
+		if (!disabled) {
+			spec.onClick();
+			onActionComplete?.();
+		}
+	}, [disabled, spec.onClick, onActionComplete]);
+
+	return (
+		<TouchableHighlight
+			style={{ ...styleSheet.button, ...activatedStyle, ...disabledStyle }}
+			onPress={onPress}
+			accessibilityLabel={ spec.accessibilityLabel }
+			accessibilityRole="button"
+			disabled={ disabled }
+		>
+			{ content }
+		</TouchableHighlight>
+	);
+};
+
+type OnToggleOverflowCallback = ()=> void;
+interface ToggleOverflowButtonProps {
+	overflowVisible: boolean;
+	onToggleOverflowVisible: OnToggleOverflowCallback;
+	styles: ToolbarStyleData;
+}
+
+// Button that shows/hides the overflow menu.
+const ToggleOverflowButton = (props: ToggleOverflowButtonProps) => {
+	const spec: ButtonSpec = {
+		icon: (
+			<MaterialIcon name="more-horiz" style={props.styles.styleSheet.text}/>
+		),
+		accessibilityLabel:
+			props.overflowVisible ? _('Hide more actions') : _('Show more actions'),
+		active: props.overflowVisible,
+		onClick: props.onToggleOverflowVisible,
+	};
+
+	return (
+		<ToolbarButton
+			styles={props.styles}
+			spec={spec}
+		/>
+	);
+};
+
+interface OverflowPopupProps {
+	buttonGroups: ButtonGroup[];
+	styles: ToolbarStyleData;
+	visible: boolean;
+
+	// Should be created using useCallback
+	onToggleOverflow: OnToggleOverflowCallback;
+}
+
+// Displays all buttons in [props.buttonGroups] if [props.visible].
+// Otherwise, displays nothing.
+const OverflowRows = (props: OverflowPopupProps) => {
+	const overflowRows: ButtonComponent[] = [];
+
+	let key = 0;
+	for (let i = 0; i < props.buttonGroups.length; i++) {
+		key++;
+		const row: ButtonComponent[] = [];
+
+		const group = props.buttonGroups[i];
+		for (let j = 0; j < group.items.length; j++) {
+			key++;
+
+			const buttonSpec = group.items[j];
+			row.push(
+				<ToolbarButton
+					key={key.toString()}
+					styles={props.styles}
+					spec={buttonSpec}
+
+					// After invoking this button's action, hide the overflow menu
+					onActionComplete={props.onToggleOverflow}
+				/>
+			);
+
+			// Show the "hide overflow" button if in the center of the last row
+			const isLastRow = i === props.buttonGroups.length - 1;
+			const isCenterOfRow = j + 1 === Math.floor(group.items.length / 2);
+			if (isLastRow && isCenterOfRow) {
+				row.push(
+					<ToggleOverflowButton
+						key={(++key).toString()}
+						styles={props.styles}
+						overflowVisible={true}
+						onToggleOverflowVisible={props.onToggleOverflow}
+					/>
+				);
+			}
+		}
+
+		overflowRows.push(
+			<View
+				key={key.toString()}
+			>
+				<ScrollView
+					horizontal={true}
+					contentContainerStyle={props.styles.styleSheet.toolbarContent}
+				>
+					{row}
+				</ScrollView>
+			</View>
+		);
+	}
+
+	if (!props.visible) {
```

Comment:
Maybe put that at the top of the function? That way you skip the loop if the component is not visible.